samples/notebooks/image-captioning-gcp/Image Captioning TF 2.0.ipynb

```diff
@@ -0,0 +1,893 @@
+    "        img = tf.image.decode_jpeg(img, channels=3)\n",
+    "        img = tf.image.resize(img, (299, 299))\n",
+    "        img = tf.keras.applications.inception_v3.preprocess_input(img)\n",
+    "        return img, image_path\n",
+    "    \n",
+    "    # Create model for processing images \n",
+    "    image_model = tf.keras.applications.InceptionV3(include_top=False,\n",
+    "                                                weights='imagenet')\n",
+    "    new_input = image_model.input\n",
+    "    hidden_layer = image_model.layers[-1].output\n",
+    "    image_features_extract_model = tf.keras.Model(new_input, hidden_layer)\n",
+    "    \n",
+    "    # Save extracted features in GCS\n",
+    "    # Get unique images\n",
+    "    encode_train = sorted(set(img_name_vector))\n",
+    "    \n",
+    "    image_dataset = tf.data.Dataset.from_tensor_slices(encode_train)\n",
+    "    image_dataset = image_dataset.map(\n",
+    "        load_image, num_parallel_calls=tf.data.experimental.AUTOTUNE).batch(batch_size)\n",
+    "    \n",
+    "    for img, path in image_dataset:\n",
+    "        batch_features = image_features_extract_model(img)\n",
+    "        batch_features = tf.reshape(batch_features,\n",
+    "                              (batch_features.shape[0], -1, batch_features.shape[3]))\n",
+    "\n",
+    "        for bf, p in zip(batch_features, path):\n",
+    "            path_of_feature = p.numpy().decode(\"utf-8\")\n",
+    "            \n",
+    "            # Save to a different location and as numpy array\n",
+    "            path_of_feature = path_of_feature.replace('.jpg', '.npy')\n",
+    "            path_of_feature = path_of_feature.replace(PATH, OUTPUT_DIR)\n",
+    "            np.save(file_io.FileIO(path_of_feature, 'w'), bf.numpy())\n",
+    "    \n",
+    "    # Create array for locations of preprocessed images\n",
+    "    preprocessed_imgs = [img.replace('.jpg', '.npy') for img in img_name_vector]\n",
+    "    preprocessed_imgs = [img.replace(PATH, OUTPUT_DIR) for img in preprocessed_imgs]\n",
+    "    \n",
+    "    # Save train_captions and preprocessed_imgs to file\n",
+    "    train_cap_path = OUTPUT_DIR + 'train_captions.npy' # array of captions\n",
+    "    preprocessed_imgs_path = OUTPUT_DIR + 'preprocessed_imgs.py'# array of paths to preprocessed images\n",
+    "    \n",
+    "    train_captions = np.array(train_captions)\n",
+    "    np.save(file_io.FileIO(train_cap_path, 'w'), train_captions)\n",
+    "    \n",
+    "    preprocessed_imgs = np.array(preprocessed_imgs)\n",
+    "    np.save(file_io.FileIO(preprocessed_imgs_path, 'w'), preprocessed_imgs)\n",
+    "    \n",
+    "    return (train_cap_path, preprocessed_imgs_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "TARGET_IMAGE = 'gcr.io/%s/preprocessing:latest' % PROJECT_NAME\n",
```

Comment:
Maybe we can extract all the TARGET_IMAGE configurations to the top cell that already contains some configurations?

Reply:
Done

Reply:
thanks